File: trifecta-android/app/src/main/java/com/belweave/trifecta/designsystem/T3Typography.kt
package com.belweave.trifecta.designsystem

import androidx.compose.ui.text.TextStyle
import androidx.compose.ui.text.font.FontFamily
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.sp

object T3Typography {
    val largeTitle: TextStyle = TextStyle(fontSize = 34.sp, fontWeight = FontWeight.Bold)
    val title: TextStyle = TextStyle(fontSize = 22.sp, fontWeight = FontWeight.SemiBold)
    val headline: TextStyle = TextStyle(fontSize = 17.sp, fontWeight = FontWeight.SemiBold)

    val body: TextStyle = TextStyle(fontSize = 16.sp, fontWeight = FontWeight.Normal)
    val bodyEmphasis: TextStyle = TextStyle(fontSize = 16.sp, fontWeight = FontWeight.SemiBold)
    val callout: TextStyle = TextStyle(fontSize = 15.sp, fontWeight = FontWeight.Normal)
    val footnote: TextStyle = TextStyle(fontSize = 13.sp, fontWeight = FontWeight.Normal)
    val caption: TextStyle = TextStyle(fontSize = 12.sp, fontWeight = FontWeight.Medium)

    val code: TextStyle = TextStyle(fontSize = 13.sp, fontFamily = FontFamily.Monospace)
    val codeBlock: TextStyle = TextStyle(fontSize = 14.sp, fontFamily = FontFamily.Monospace)
}

File: trifecta-ios/Trifecta/DesignSystem/T3Typography.swift
import SwiftUI
import UIKit

enum T3Typography {
    static func display(size: CGFloat, weight: Font.Weight = .semibold) -> Font {
        if let _ = UIFont(name: "DMSans-Regular", size: size) {
            let psName: String
            switch weight {
            case .bold, .heavy, .black: psName = "DMSans-Bold"
            case .semibold:             psName = "DMSans-SemiBold"
            case .medium:               psName = "DMSans-Medium"
            default:                    psName = "DMSans-Regular"
            }
            return Font.custom(psName, size: size)
        }
        return Font.system(size: size, weight: weight, design: .default)
    }

    static let largeTitle = display(size: 34, weight: .bold)
    static let title      = display(size: 22, weight: .semibold)
    static let headline   = display(size: 17, weight: .semibold)

    static let body         = display(size: 16, weight: .regular)
    static let bodyEmphasis = display(size: 16, weight: .semibold)
    static let callout      = display(size: 15, weight: .regular)
    static let footnote     = display(size: 13, weight: .regular)
    static let caption      = display(size: 12, weight: .medium)

    static let code         = Font.system(.footnote, design: .monospaced)
    static let codeBlock    = Font.system(.callout,  design: .monospaced)
}
